Sources/StrongContractResponse/All/Vapor.Request.swift

```diff
@@ -18,18 +18,9 @@ extension Vapor.Request {
         return try encryptedData.decrypt()
     }
 
-    func codableBody<T: Codable>() throws -> T {
-        guard let bodyData = body.data else { throw GenericError(text: "Body data was nil") }
-        return try JSONDecoder().decode(T.self, from: bodyData.data)
-    }
-
-    func codableBodyExemptingData<T: Codable>() throws -> T {
-        if T.self == Data.self, let data = body.data?.data as? T {
-            return data
-        } else if let bodyData = body.data {
-            return try JSONDecoder().decode(T.self, from: bodyData.data)
-        }
-        throw GenericError(text: "Body data was nil")
+    func decodedObject<T: Decodable>(using decoder: JSONDecoder = .init()) throws -> T {
+        guard let data = body.data?.data else { throw GenericError(text: "Body data was nil") }
+        return try data.decodedObject(using: decoder)
```

Comment:
This would default to encoding and decoding data...this would cause the issue you raised awareness about...What are you doing?

Reply:
This is what decodes the payload not what is returned

Reply:
And if the payload is Data, then we shouldn't decode it right?

Reply:
Because we don't run decode and encode to convert Data to Data, it is already data.

Reply:
What is wrong with this again? 

If T is Data, then it simply returns data without decoding or encoding.  If T is not Data, then it tries to decode it to the Codable type.  Thats what we want right?

Reply:
This will never work. It returns a Codable. A Codable means JSON which is always text.

Reply:
Doesn't it return Decodable? So you're saying, even if we don't call encode nor decode, it will still convert the Data to a String "under the hood"?

Reply:
probably yes if content type is JSON

Reply:
Again here we are just decoding the payload

Reply:
Probably? 
I thought the issue came when we call encode and decode.